### PR TITLE
Add String#center method

### PR DESCRIPTION
**Pull request by hutou**
`String#center(width, char) `returns a new string of length `width` with original string centered, and padded with `char` (default = space)

### CODE REVIEW

File: src/string.cr [hunk 1]
```diff
@@ -3627,37 +3627,61 @@ class String
...
+          leftpadding.times do
+            Intrinsics.memcpy(buffer.as(Void*), bytes.to_unsafe.as(Void*), count, false)
+            buffer += count
+          end
         end
       end
-
-      unless left
-        buffer.copy_from(to_unsafe, bytesize)
+      Intrinsics.memcpy(buffer.as(Void*), to_unsafe.as(Void*), bytesize, false)
```
Comment (wontruefree) [inline src/string.cr]: I think you might be able to use `Pointer.copy_from`

```suggestion
      buffer.copy_from(to_unsafe, bytesize)
```

Comment (wontruefree) [reply src/string.cr]: I see now this is diff noise so this might be outside the scope of this PR.

**Comment by straight-shoota** [comment]
This continues the previous PR #3924 (stalled).

File: src/string.cr (hunk 1 above)
Comment (hutou) [reply src/string.cr]: Is this "better" than memcpy ?

File: src/string.cr [hunk 2]
```diff
@@ -3627,37 +3627,61 @@ class String
...
-    new_bytesize = bytesize + difference * count
+    padding = (len - size)
+    new_bytesize = bytesize + padding * count
+    case justify
+    when .< 0
+      leftpadding, rightpadding = 0, padding
+    when .> 0
+      leftpadding, rightpadding = padding, 0
+    else
+      leftpadding = padding >> 1
```
Comment (RX14) [inline src/string.cr]: `// 2`, please

there's no need for manual pinhole optimizations the compiler is perfectly capable of.

File: src/string.cr (hunk 1 above)
Comment (RX14) [reply src/string.cr]: It's the same as `memcpy`, but shorter.

It's fine to leave this as-is for this PR.

File: src/string.cr (hunk 2 above)
Comment (hutou) [reply src/string.cr]: Ok, done !

**Comment by RX14** [comment]
Thank you!

**Pull request merged**